src/views/Farms/components/SearchInput.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useState, useRef } from 'react'
+import { Input, SearchIcon, IconButton } from '@pancakeswap-libs/uikit'
+import styled, { css } from 'styled-components'
+
+const StyledInput = styled(Input)`
+  border-radius: 1rem;
+  padding-right: 5rem;
+  transition: 0.3s;
+  margin-left: auto;
+  ${({ theme }) => theme.mediaQueries.sm} {
+    width: 100% !important;
+    padding: 0 1rem !important;
+    opacity: 1 !important;
+  }
+`
+
+const InputWrapper = styled.div`
+  position: relative;
+  ${({ theme }) => theme.mediaQueries.sm} {
+    width: 20rem;
+    display: block;
+  }
+`
+
+const SearchButton = styled.button`
+  background: transparent;
+  border: none;
+  outline: none;
+  position: absolute;
+  right: 0.25rem;
+  top: 50%;
+  transform: translate(0px, -50%);
+  display: flex;
+  align-items: center;
+  cursor: pointer;
+  opacity: 0;
+  transition: 0.3s;
+
+  ${({ theme }) => theme.mediaQueries.sm} {
+    opacity: 1 !important;
+  }
+`
+
+const Container = styled.div<{ toggled: boolean }>`
+  margin-left: auto;
+  position: absolute;
+  right: 0;
+
+  ${(props) => css`
+    left: ${props.toggled ? '0' : 'auto'};
+    ${StyledInput} {
+      width: ${props.toggled ? '100%' : '0'};
+      padding: ${props.toggled ? '0 1rem' : '0'};
+      opacity: ${props.toggled ? '1' : '0'};
+    }
+
+    ${SearchButton} {
+      opacity: ${props.toggled ? '1' : '0'};
+    }
+  `}
+
+  ${({ theme }) => theme.mediaQueries.sm} {
+    left: auto;
+  }
+`
+
+const Label = styled.span`
+  margin-right: 0.375rem;
+  color: ${(props) => props.theme.colors.primary};
+  font-size: 1rem;
+  font-weight: 600;
+`
+
+const MobileContainer = styled.div`
+  display: block;
+  position: absolute;
+  right: 0;
+  top: 50%;
+  transform: translate(0, -50%);
+
+  ${({ theme }) => theme.mediaQueries.sm} {
+    display: none;
+  }
+`
+
+interface Props {
+  value: string
+  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
```

Comment:
Please use `evt` or `event`. We want to avoid one letter variables even if it is a widely known one.